### ISSUE TITLE
After converting .pt file into .ptl (PyTorch lite) the file size is not decreasing.

### ISSUE THREAD

**Issue by mansi733** [ultralytics/yolov5#7272]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

![saa](https://user-images.githubusercontent.com/82514056/161512941-40602828-f5be-4701-8814-790c09fb143a.jpg)

**Comment by glenn-jocher** [comment]
@mansi733 why would you expect file sizes to decrease on export?

**Comment by mansi733** [comment]
I thought because it is in lite format and we are optimizing it while exporting, please correct me if I am wrong.
Also, after conversion, I am losing accuracy.
e.g - In YOLOv5, the model is able to detect all the objects but after conversion, the model is not detecting all the objects in android.

**Comment by glenn-jocher** [comment]
@mansi733 YOLOv5 models are benchmarked across all export formats already. Results show no loss in accuracy across any format:

Run YOLOv5 benchmarks (speed and accuracy) for all supported export formats. This PR adds GPU benchmarking capability following CPU benchmarking PR https://github.com/ultralytics/yolov5/pull/6613.

Format                      | `export.py --include`         | Model
:---                        | --:                           | :--
[PyTorch](https://pytorch.org/)                     | -                             | `yolov5s.pt`
[TorchScript](https://pytorch.org/docs/stable/jit.html)                 | `torchscript`                 | `yolov5s.torchscript`
[ONNX](https://onnx.ai/)                        | `onnx`                        | `yolov5s.onnx`
[OpenVINO](https://docs.openvino.ai/latest/index.html)                    | `openvino`                    | `yolov5s_openvino_model/`
[TensorRT](https://developer.nvidia.com/tensorrt)                    | `engine`                      | `yolov5s.engine`
[CoreML](https://github.com/apple/coremltools)                      | `coreml`                      | `yolov5s.mlmodel`
[TensorFlow SavedModel](https://www.tensorflow.org/guide/saved_model)       | `saved_model`                 | `yolov5s_saved_model/`
[TensorFlow GraphDef](https://www.tensorflow.org/api_docs/python/tf/Graph)         | `pb`                          | `yolov5s.pb`
[TensorFlow Lite](https://www.tensorflow.org/lite)             | `tflite`                      | `yolov5s.tflite`
[TensorFlow Edge TPU](https://coral.ai/docs/edgetpu/models-intro/)         | `edgetpu`                     | `yolov5s_edgetpu.tflite`
[TensorFlow.js](https://www.tensorflow.org/js)               | `tfjs`                        | `yolov5s_web_model/`

Usage: 
```python
git clone https://github.com/ultralytics/yolov5 -b update/bench_gpu  # clone
cd yolov5
pip install -qr requirements.txt coremltools onnx onnxruntime-gpu openvino-dev  # install
pip install -U nvidia-tensorrt --index-url https://pypi.ngc.nvidia.com  # TensorRT

python utils/benchmarks.py --weights yolov5s.pt --img 640 --device 0
```

### Colab++ V100 High-RAM Results

```
benchmarks: weights=/content/yolov5/yolov5s.pt, imgsz=640, batch_size=1, data=/content/yolov5/data/coco128.yaml, device=0, half=False
Checking setup...
YOLOv5 🚀 v6.1-48-g0c1025f torch 1.10.0+cu111 CUDA:0 (Tesla V100-SXM2-16GB, 16160MiB)
Setup complete ✅ (8 CPUs, 51.0 GB RAM, 46.1/166.8 GB disk)

Benchmarks complete (433.63s)
                   Format  mAP@0.5:0.95  Inference time (ms)
0                 PyTorch      0.462296             9.159939
1             TorchScript      0.462296             6.607546
2                    ONNX      0.462296            12.698026
3                OpenVINO           NaN                  NaN
4                TensorRT      0.462280             1.725197
5                  CoreML           NaN                  NaN
6   TensorFlow SavedModel      0.462296            20.273019
7     TensorFlow GraphDef      0.462296            20.212173
8         TensorFlow Lite           NaN                  NaN
9     TensorFlow Edge TPU           NaN                  NaN
10          TensorFlow.js           NaN                  NaN
```

Note: TensorRT exports are fixed at FP16.

**Comment by glenn-jocher** [comment]
### Ultralytics Hyperplane EPYC Milan AMD CPU Results
```
benchmarks: weights=/usr/src/app/yolov5s.pt, imgsz=640, batch_size=1, data=/usr/src/app/data/coco128.yaml
Checking setup...
/bin/sh: 1: git: not found
YOLOv5 🚀 2022-3-11 torch 1.11.0+cpu CPU
Setup complete ✅ (96 CPUs, 1007.7 GB RAM, 885.4/3519.3 GB disk)

Benchmarks complete (334.78s)
                   Format  mAP@0.5:0.95  Inference time (ms)
0                 PyTorch      0.462296            59.022402
1             TorchScript      0.462296            81.963500
2                    ONNX      0.462296            66.349735
3                OpenVINO      0.462296            28.817065
4                TensorRT           NaN                  NaN
5                  CoreML           NaN                  NaN
6   TensorFlow SavedModel      0.462296            91.322277
7     TensorFlow GraphDef      0.462296            96.317230
8         TensorFlow Lite      0.462334           160.701267
9     TensorFlow Edge TPU           NaN                  NaN
10          TensorFlow.js           NaN                  NaN
```

**Comment by mansi733** [comment]
![and1](https://user-images.githubusercontent.com/82514056/161552728-f70b66e8-f1ae-4f76-a9d7-618ef9097caa.jpg)
![yolo1](https://user-images.githubusercontent.com/82514056/161552767-ee9e2757-782d-43c2-a72e-07666ff7a76b.jpg)
 
the first image is the output of android studio and the other one is from the YOLOv5 medium version.

**Comment by glenn-jocher** [comment]
@mansi733 well, there is known degradation for TFLite INT8 models (results above shown for TFLite FP16). But if you are using TFLite FP16 then any differences with PyTorch are due to your Android implementation rather than the export process, which as we've shown above validates no accuracy loss.

**Comment by mansi733** [comment]
Okay, I was using Torchscript to generate the .ptl format.
We just need to upload the .ptl file in android studio, please correct me if I am doing anything wrong.

**Comment by glenn-jocher** [comment]
@mansi733 you are on the right track. Converting your model to TorchScript and then deploying the .ptl file to Android Studio is the correct approach. Ensure that your Android implementation is correctly loading and using the .ptl model for inference. If you encounter any issues, please refer to the Ultralytics Docs for YOLOv5 integration with Android Studio: https://docs.ultralytics.com/yolov5/
